src/record_layout.zig

```diff
@@ -0,0 +1,459 @@
+const std = @import("std");
+const Type = @import("Type.zig");
+const Attribute = @import("Attribute.zig");
+const Compilation = @import("Compilation.zig");
+const Parser = @import("Parser.zig");
+const Record = Type.Record;
+const Field = Record.Field;
+const TypeLayout = Type.TypeLayout;
+
+// almost all the code for record layout
+// was liberally copied from this repro
+// https://github.com/mahkoh/repr-c
+// specifically the code in "repc/impl/src/" directory.
+// At the time of this writing (Q3 2022)
+// that code marked as "SPDX-License-Identifier: MIT OR Apache-2.0"
+// so is compatible with arocc's MIT licence
+
+const BITS_PER_BYTE: u29 = 8;
```

Comment:
```suggestion
const BITS_PER_BYTE = 8;
```
No types on constants. Also lower snake case would be more conventional but I don't care about that too much.